client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx

```diff
@@ -0,0 +1,208 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import { localize } from 'i18n-calypso';
+import { isEmpty } from 'lodash';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FoldableCard from 'components/foldable-card';
+import SectionHeader from 'components/section-header';
+import { checkInboundTransferStatus } from 'lib/domains';
+
+class TransferDomainPrecheck extends React.PureComponent {
+	static propTypes = {
+		domain: PropTypes.string,
+		setValid: PropTypes.func,
+	};
+
+	state = {
+		unlocked: false,
+		privacy: false,
+		email: '',
+		loading: true,
+	};
+
+	componentWillMount() {
+		this.refreshStatus();
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( nextProps.domain !== this.props.domain ) {
+			this.refreshStatus();
+		}
+	}
+
+	onClick = () => {
+		this.props.setValid( this.props.domain );
+	};
+
+	refreshStatus = () => {
+		this.setState( { loading: true } );
+
+		checkInboundTransferStatus( this.props.domain, ( error, result ) => {
+			if ( ! isEmpty( error ) ) {
+				return;
+			}
+
+			this.setState( {
+				email: result.admin_email,
+				privacy: false,
+				unlocked: result.unlocked,
+				loading: false,
+			} );
+		} );
+	};
+
+	getSection( heading, message, explanation, position, button ) {
+		const header = (
+			<div>
+				<div className="transfer-domain-step__section-heading">
+					<span className="transfer-domain-step__section-heading-number">{ position }</span>
+					{ heading }
+				</div>
+				<div>
+					<small>{ message }</small>
+				</div>
+			</div>
+		);
+
+		const cardClasses = classNames( {
+			'hide-secondary': ! button,
+		} );
+
+		return (
+			<FoldableCard
+				className={ cardClasses }
+				header={ header }
+				summary={ button }
+				expandedSummary={ button }
+			>
+				{ explanation }
+			</FoldableCard>
+		);
+	}
+
+	getStatusMessage() {
+		const { translate } = this.props;
+		const { unlocked } = this.state;
+
+		const heading = unlocked
+			? translate( 'Domain is unlocked.' )
+			: translate( 'Unlock the domain.' );
+		const message = unlocked
+			? translate( 'Your domain is unlocked at your current registrar.' )
+			: translate(
+					"You'll need to unlock the domain at your current registrar before we can move it."
+				);
+		const explanation = translate(
```

Comment:
This leads to a confusing state, when the domain is unlocked, but the explanation is shown:
<img width="740" alt="screen shot 2017-11-14 at 21 03 10" src="https://user-images.githubusercontent.com/3392497/32802108-51316d9c-c97f-11e7-93f9-91748481c157.png">